### PR TITLE
Fixes #35231 - Add testing support for ouiaIds

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

1. Adds a `checkforOuiaIds` function in `global_test_setup.js`. When this is uncommented properly, your tests will still pass but will alert you of any PF components where you forgot to add an `ouiaId` prop.

```
[vagrant@centos8-katello-devel katello]$ npx jest webpack/components/extensions/HostDetails/Tabs/__tests__/repositorySetsTab.test.js
  console.error webpack/global_test_setup.js:9
    Warning: Failed prop type: The prop `ouiaId` is marked as required in `Dropdown`, but its value is `undefined`.
        in Dropdown (created by TypeAheadItems)
        in TypeAheadItems (created by TypeAheadSearch)
        in TypeAheadSearch (created by Downshift)
        in div (created by Downshift)
        in Downshift (created by TypeAhead)
        in TypeAhead (created by Search)
        in div (created by Search)
        in Search (created by Connect(Search))
        in Connect(Search) (created by TableWrapper)
        in div (created by FlexItem)
        in FlexItem (created by TableWrapper)
        in div (created by Flex)
        in Flex (created by TableWrapper)
        in TableWrapper (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in RepositorySetsTab
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

  console.error webpack/global_test_setup.js:9
    Warning: Failed prop type: The prop `ouiaId` is marked as required in `Pagination`, but its value is `undefined`.
        in Pagination (created by Pagination)
        in Pagination (created by PageControls)
        in div (created by FlexItem)
        in FlexItem (created by PageControls)
        in PageControls (created by TableWrapper)
        in div (created by Flex)
        in Flex (created by TableWrapper)
        in TableWrapper (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in RepositorySetsTab
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

  console.error webpack/global_test_setup.js:9
    Warning: Failed prop type: The prop `ouiaId` is marked as required in `DropdownItem`, but its value is `undefined`.
        in DropdownItem (created by PaginationOptionsMenu)
        in PaginationOptionsMenu (created by Pagination)
        in div (created by Pagination)
        in Pagination (created by Pagination)
        in Pagination (created by PageControls)
        in div (created by FlexItem)
        in FlexItem (created by PageControls)
        in PageControls (created by TableWrapper)
        in div (created by Flex)
        in Flex (created by TableWrapper)
        in TableWrapper (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in RepositorySetsTab
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

  console.error webpack/global_test_setup.js:9
    Warning: Failed prop type: The prop `ouiaId` is marked as required in `DropdownToggle`, but its value is `undefined`.
        in DropdownToggle (created by OptionsToggle)
        in OptionsToggle (created by PaginationOptionsMenu)
        in div (created by Context.Consumer)
        in DropdownWithContext (created by PaginationOptionsMenu)
        in PaginationOptionsMenu (created by Pagination)
        in div (created by Pagination)
        in Pagination (created by Pagination)
        in Pagination (created by PageControls)
        in div (created by FlexItem)
        in FlexItem (created by PageControls)
        in PageControls (created by TableWrapper)
        in div (created by Flex)
        in Flex (created by TableWrapper)
        in TableWrapper (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in RepositorySetsTab
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

  console.error webpack/global_test_setup.js:9
    Warning: Failed prop type: The prop `ouiaId` is marked as required in `Button`, but its value is `undefined`.
        in Button (created by Navigation)
        in Navigation (created by Pagination)
        in div (created by Pagination)
        in Pagination (created by Pagination)
        in Pagination (created by PageControls)
        in div (created by FlexItem)
        in FlexItem (created by PageControls)
        in PageControls (created by TableWrapper)
        in div (created by Flex)
        in Flex (created by TableWrapper)
        in TableWrapper (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in div (created by RepositorySetsTab)
        in RepositorySetsTab
        in Router (created by MemoryRouter)
        in MemoryRouter
        in Provider

 PASS  webpack/components/extensions/HostDetails/Tabs/__tests__/repositorySetsTab.test.js (6.956s, 216 MB heap size)
  ✓ Can call API for repository sets and show basic table (245ms)
  ✓ Can handle no repository sets being present (75ms)
  ✓ Toggle Group shows if it's not the default content view or library enviroment (116ms)
  ✓ Toggle Group shows if it's the default content view but non-library environment (105ms)
  ✓ Toggle Group shows if it's the library environment but a non-default content view (109ms)
  ✓ Toggle Group does not show if it's the library environment and default content view (85ms)
  ✓ Can toggle with the Toggle Group  (216ms)
  ✓ Can override to disabled (198ms)
  ✓ Can override to enabled (174ms)
  ✓ Can reset to default (159ms)
  ✓ Can override in bulk (682ms)
  ✓ Can filter by status (237ms)
  ✓ Can display osRestricted as a label (88ms)

Test Suites: 1 passed, 1 total
Tests:       13 passed, 13 total
```

Note that each error displays the React component tree, so it should be easy to find the correct JS file and add the prop.
3. Adds `ouiaId`s in `RepositorySetsTab.js`.

#### Considerations taken when implementing this change?

If a component is imported from `foremanReact`, it will still flag it if the ouiaId is missing. So you may not be able to eliminate all ouiaId issues in the Katello code base alone.

#### What are the testing steps for this pull request?

* Follow the instructions I added to `global_test_setup.js`
* Run any test file: `npx jest <path_to_file>.js`

### CODE REVIEW

**Comment by jeremylenz** [comment]
> Do the ouia-id changes in tests reflect all missing ouia-ids currently?

When uncommented per the instructions, I believe so.  The exception is that if a component doesn't have test coverage, this check wouldn't work.

File: webpack/ouia_id_check.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    Tr,
+  ];
+  beforeEach(() => {
+    // eslint-disable-next-line no-restricted-syntax
+    for (const Component of ouiaSupportedPFComponents) {
+      // eslint-disable-next-line no-continue
+      if (!Component) continue;
+      Component.propTypes = {
+        ...Button.propTypes,
+        ouiaId: PropTypes.string.isRequired,
```
Comment (sjha4) [inline webpack/ouia_id_check.js:89]: `...Button.propTypes,` why is this needed?

Comment (jeremylenz) [reply webpack/ouia_id_check.js:89]: Oops.. That should be `Component`. Will fix
